### PR TITLE
FreshDataApi: Add mutations

### PR DESCRIPTION
**Pull request by coderkevin**
This adds mutations to the api as a set of purpose-made mutations for
applications to use to send mutating operations to an API.

To Test:
1. `npm install`
2. `npm test`

No example code for testing mutations yet, but I'll be adding it to the wp plugin example next.

### CODE REVIEW

File: src/client/index.js [hunk 1]
```diff
@@ -201,9 +206,16 @@ function mapMethods( methods, clientKey ) {
...
+		return mappedMutations;
+	}, {} );
+}
+
 function mapSelectors( selectors, clientGetData, clientRequireData ) {
 	return Object.keys( selectors ).reduce( ( mappedSelectors, selectorName ) => {
 		mappedSelectors[ selectorName ] = selectors[ selectorName ]( clientGetData, clientRequireData );
 		return mappedSelectors;
-	}, [] );
+	}, {} );
```
Comment (timmyc) [inline src/client/index.js]: Curious about this change here - is it just to make it the same as mutations being stored in an object?

Comment (coderkevin) [reply src/client/index.js]: It was actually incorrect, and I didn't notice it before.

**Comment by coderkevin** [comment]
I changed the parameters passed to the mutations to be functional operations for create, update, and delete. This should be more straightforward.

I did notice that this reduces code coverage by a bit, but I will submit another patch splitting out the test noted by the TODO in the code here for that.

File: src/react-redux/__tests__/with-api-client.spec.js [hunk 2]
```diff
@@ -227,5 +237,95 @@ describe( 'withApiClient', () => {
...
+		it( 'should call mapMutationsToProps if present.', () => {
+			mapMutationsToProps = jest.fn();
+
+			const ComponentWithApiClient = withApiClient( 'test', { mapMutationsToProps, getClientKey } )( Component );
+			mount( <ComponentWithApiClient clientKey="123" />, { context: { getApiClient } } );
+
+			expect( mapMutationsToProps ).toHaveBeenCalledTimes( 1 );
+		} );
+
+		// TODO: This test runs code a bit deep, split it up into an ApiClient test and this one.
```
Comment (timmyc) [inline src/react-redux/__tests__/with-api-client.spec.js]: It is a bit deep of a test, but pretty nice to see it all written out here - great example code.

**Comment by coderkevin** [comment]
Thanks for the feedback! I'm glad to have some real-world code put on this to test the abstractions. It's very helpful.

**Pull request merged**